keps/sig-node/3066-subsecond-probes/README.md

```diff
@@ -0,0 +1,859 @@
+- Complete a performance analysis/review and adjust
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- Collect examples of real-world usage
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+### Upgrade / Downgrade Strategy
+
+Design will be tested to be hardened against upgrade/downgrade issues.
+
+### Version Skew Strategy
+
+Design consideration has been made for and will be tested to ensure forward and backward version skew compatibility.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ X ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: SubSecondProbes
+    - Components depending on the feature gate:
+		kube-apiserver and kubelet
+
+* **Does enabling the feature change any default behavior?**
+
+  No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+  Yes, see the [drop test](https://github.com/psschwei/kubernetes/blob/1ca40771e26b96d6121c49b19c2cbe6466694c7a/pkg/api/pod/util_test.go#L1029)
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+  No breaking changes
+
+###### Are there any tests for feature enablement/disablement?
+
+  Yes, see the [drop test](https://github.com/psschwei/kubernetes/blob/1ca40771e26b96d6121c49b19c2cbe6466694c7a/pkg/api/pod/util_test.go#L1029)
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+TBD, no failures identified as possible at this time
+
+###### What specific metrics should inform a rollback?
+
+TBD
```

Comment:
I'd suggest thinking through metrics early on, to simplify the decision that will drive rollback at later stages.

Reply:
ok.. in general any metrics showing an unexpected degradation of resource utilization
================================================================
container_cpu_usage_seconds_total
container_memory_working_set_bytes
container_start_time_seconds
node_cpu_usage_seconds_total
node_memory_working_set_bytes
pod_cpu_usage_seconds_total
pod_memory_working_set_bytes
scheduler_pending_pods
scheduler_pod_scheduling_attempts
scheduler_pod_scheduling_duration_seconds
scheduler_preemption_attempts_total
scheduler_preemption_victims
scheduler_queue_incoming_pods_total
scheduler_schedule_attempts_total
scheduler_scheduling_attempt_duration_seconds
kubelet_*all resource utilization and error processing related metrics
node_*    ""     ""
pod_swap_usage_bytes
prober_probe_duration_seconds
prober_probe_total


..

Reply:
I'd suggest calling out a few, explaining what cluster admin should look for when trying to better understand issues with this functionality.

Reply:
ok..